### PR TITLE
PUD-1380 - don't default details in 'add missing documents record' form

### PR DESCRIPTION
**Pull request by jonnywyatt**

### CODE REVIEW

File: integration_tests/integration/uploaded-documents-missing.spec.js [hunk 1]
```diff
@@ -103,37 +102,6 @@ context('Missing uploaded documents', () => {
     })
   })
```
Comment (philsegal) [inline integration_tests/integration/uploaded-documents-missing.spec.js:104]: do we need a test to verify that the entry is now empty upon previous mdr?

File: server/controllers/utils/getFormValues.ts [hunk 2]
```diff
@@ -112,7 +112,7 @@ export const getFormValues = ({ errors = {}, unsavedValues = {}, apiValues }: Ge
   // missing documents detail
   values.missingDocumentsDetail = isDefined(errors.missingDocumentsDetail)
     ? ''
-    : unsavedValues.missingDocumentsDetail || apiValues.missingDocumentsRecords?.[0]?.details
+    : unsavedValues.missingDocumentsDetail || undefined
```
Comment (philsegal) [inline server/controllers/utils/getFormValues.ts:115]: why is this here rather than in the above block with the other 'standard' fields?

Comment (jonnywyatt) [reply server/controllers/utils/getFormValues.ts:115]: It's a distinct case, there's no intention of using the API value, which is what the other clauses do

Comment (philsegal) [reply server/controllers/utils/getFormValues.ts:115]: I see, because we can never edit an MDR? Should the note detail not also be here? Rescind detail? And other unchangeable fields?

File: integration_tests/integration/uploaded-documents-missing.spec.js (hunk 1 above)
Comment (jonnywyatt) [reply integration_tests/integration/uploaded-documents-missing.spec.js:104]: Added an integration test

File: server/controllers/utils/getFormValues.ts (hunk 2 above)
Comment (jonnywyatt) [reply server/controllers/utils/getFormValues.ts]: Notes, rescind records and missing doc records now handled the same way

**Review by philsegal — APPROVED** [review]
LGTM 🤪
